File: Math/codeforces_271A.py

# Problem link ===>> https://codeforces.com/problemset/problem/271/A

def is_beautiful(year):
  """Returns True if the given year is beautiful, False otherwise."""
  digits = set()
  for digit in str(year):
    digits.add(digit)
  return len(digits) == 4

def find_next_beautiful_year(year):
  """Returns the next beautiful year after the given year."""
  while not is_beautiful(year):
    year += 1
  return year

def main():
  year = int(input())
  next_beautiful_year = find_next_beautiful_year(year)
  print(next_beautiful_year)

if __name__ == '__main__':
  main()

File: Math/codeforces_271A.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/271/A
// submission link ===>> https://codeforces.com/contest/271/submission/225685773

#include <iostream>
using namespace std;

int main()
{
    int year;
    cin >> year;

    while (true)
    {
        year++;

        // Get the individual digits of the year.
        int a = year / 1000;
        int b = (year / 100) % 10;
        int c = (year / 10) % 10;
        int d = year % 10;

        // Check if all the digits are unique.
        if (a != b && a != c && a != d && b != c && b != d && c != d)
        {
            break;
        }
    }

    cout << year << endl;

    return 0;
}
